### PR TITLE
Update some features

### PR DESCRIPTION
**Pull request by asdflj**
Update some features
1.click record to rename P2P Tunnel (Help player to build large ae network)
2.shift click record. Well be transport player to P2P Tunnel position (OP only)
@GlodBlock   https://github.com/GTNewHorizons/BetterP2P/pull/11 fix when modify name, P2P Tunnel device context not change

image
image

### CODE REVIEW

**Comment by asdflj** [comment]
简单的来说就是刚刚的问题修好了
你觉得重命名按钮多大合适?
另外我的英语不太行
7N$05WJH$~~G$QE~FM7WH~A

File: src/main/java/com/projecturanus/betterp2p/network/ServerRenameP2PTunnel.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import net.minecraftforge.common.DimensionManager
+import net.minecraftforge.common.util.ForgeDirection
+
+class ServerRenameP2PTunnel : IMessageHandler<C2SP2PTunnelInfo, IMessage?> {
+    override fun onMessage(message: C2SP2PTunnelInfo, ctx: MessageContext): IMessage? {
+        val world: World =  DimensionManager.getWorld(message.info.world)
+        val te: TileEntity = world.getTileEntity(message.info.posX,message.info.posY,message.info.posZ)
+        val player = ctx.serverHandler.playerEntity
+        if (te is IGridHost && te.getGridNode(ForgeDirection.getOrientation(message.info.side)) != null) {
+            val p = (te as IPartHost).getPart(ForgeDirection.getOrientation(message.info.side)) as  PartP2PTunnel<*>
```
Comment (GlodBlock) [inline src/main/java/com/projecturanus/betterp2p/network/ServerRenameP2PTunnel.kt:22]: this line cause NPE when i testing

Comment (asdflj) [reply src/main/java/com/projecturanus/betterp2p/network/ServerRenameP2PTunnel.kt:22]: solved NPE issues 
7N$05WJH$~~G$QE~FM7WH~A
话说咋下面的这个提交怎么合并起来 XD

Comment (GlodBlock) [reply src/main/java/com/projecturanus/betterp2p/network/ServerRenameP2PTunnel.kt:22]: 哪个？

Comment (asdflj) [reply src/main/java/com/projecturanus/betterp2p/network/ServerRenameP2PTunnel.kt:22]: 下面的两个提交

Comment (GlodBlock) [reply src/main/java/com/projecturanus/betterp2p/network/ServerRenameP2PTunnel.kt:22]: 你交在一个分支里就自动合并了

Comment (asdflj) [reply src/main/java/com/projecturanus/betterp2p/network/ServerRenameP2PTunnel.kt:22]: L_RUX~IJ)P98RKNOE%0~D%E

Comment (asdflj) [reply src/main/java/com/projecturanus/betterp2p/network/ServerRenameP2PTunnel.kt:22]: thanks a lot

**Review by GlodBlock — APPROVED** [review]
should be fine now

**Pull request merged**